pkg/kotsclient/customer_list_by_app_version.go

```diff
@@ -0,0 +1,56 @@
+package kotsclient
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/replicated/pkg/types"
+)
+
+type CustomerListWithInstancesResponse struct {
+	// Response from /v3/app/{appID}/customers
+	Customers      []types.Customer `json:"customers"`
+	TotalCustomers int              `json:"totalCustomers"`
+}
+
+func (c *VendorV3Client) ListCustomersByAppAndVersion(appID string, appVersion string, appType string) ([]types.Customer, error) {
+
+	matchingCustomers := []types.Customer{}
+	page := 0
+	for {
+		resp := CustomerListWithInstancesResponse{}
+		path := fmt.Sprintf("/v3/app/%s/customers?currentPage=%d", appID, page)
+		err := c.DoJSON("GET", path, http.StatusOK, nil, &resp)
+		if err != nil {
+			return nil, errors.Wrapf(err, "List Customers By App Version page %d", page)
+		}
+
+		for _, customer := range resp.Customers {
+			for _, instance := range customer.Instances {
+				for versionIndex, versionHistory := range instance.VersionHistory {
+					if versionHistory.VersionLabel == appVersion && versionIndex == 0 {
+						// only add the customer if the version matches
+						// index[0] is the latest version
+						// There has to be a better way to do this without creating a tempCustomer
+						tempCustomer := customer
+						tempCustomer.Instances = []types.Instance{instance}
+						tempCustomer.Instances[0].VersionHistory = []types.VersionHistory{versionHistory}
+						matchingCustomers = append(matchingCustomers, tempCustomer)
+
+					}
+				}
+			}
+		}
+		if len(matchingCustomers) == resp.TotalCustomers || len(resp.Customers) == 0 {
+			// We've reached the end of the customer pages
+			break
+		}
+		page = page + 1
+	}
+
+	if len(matchingCustomers) == 0 {
+		return matchingCustomers, fmt.Errorf("no matching customers found for app %q and version %q ", appID, appVersion)
```

Comment:
if someone has `--output json`, will this throw an error instead of the expected return of `[]` ?

Reply:
I addressed this in cli/cmd/customer_ls.go
When customers is 0 and output is json, we pass a blank struct to the print function, which will result in []
Otherwise, the output format is table, in which case, we raise an error instead.

// if err and outputFormat is json, customers should be a blank struct and print will return []
		if err != nil && r.outputFormat == "json" {
			return print.CustomersWithInstances(r.outputFormat, r.w, customers)